src/request/sign-polygon-transaction/SignPolygonTransactionApi.js

```diff
@@ -95,10 +95,10 @@ class SignPolygonTransactionApi extends PolygonRequestParserMixin(TopLevelApi) {
             if (!['transfer', 'transferWithPermit'].includes(description.name)) {
                 throw new Errors.InvalidRequestError('Requested Polygon contract method is invalid');
             }
-        } else if (forwardRequest.to === CONFIG.BRIDGED_USDC_HTLC_CONTRACT_ADDRESS) {
+        } else if (forwardRequest.to === CONFIG.NATIVE_USDC_HTLC_CONTRACT_ADDRESS) {
             const usdcHtlcContract = new ethers.Contract(
-                CONFIG.BRIDGED_USDC_HTLC_CONTRACT_ADDRESS,
-                PolygonContractABIs.BRIDGED_USDC_HTLC_CONTRACT_ABI,
+                CONFIG.NATIVE_USDC_HTLC_CONTRACT_ADDRESS,
+                PolygonContractABIs.NATIVE_USDC_HTLC_CONTRACT_ABI,
             );
 
             /** @type {PolygonRefundDescription} */
```

Comment:
Could potentially check here that the `target` address is the user's address.
Not sure whether it's checked somewhere.
The UI displays the address but also the key label (without checking that it's actually the correct address for that label). Thus users that only look at the label would miss an address mismatch.
Obviously users should check the address, but adding a check here would probably also not hurt.

Reply:
we can only check the address after the password input, as only then it's derived.

I have added a check for this in the password confirm handler.